### PR TITLE
Add a smoke test to detect unexpected otel metrics

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DetectUnexpectedOtelMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+abstract class DetectUnexpectedOtelMetricsTest {
+
+  private static final Set<String> EXPECTED_OTEL_METRICS = new HashSet<>(4);
+
+  static {
+    EXPECTED_OTEL_METRICS.add("http.server.duration"); // Servlet
+    EXPECTED_OTEL_METRICS.add("http.client.duration"); // HttpClient
+    EXPECTED_OTEL_METRICS.add("rpc.client.duration"); // gRPC
+    EXPECTED_OTEL_METRICS.add("rpc.server.duration"); // gRPC
+  }
```
Comment (trask) [inline smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DetectUnexpectedOtelMetricsTest.java:40]: are these expected? I thought we transformed these into "standard breeze metrics"

File: smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/resources/applicationinsights.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "connectionStringOverrides": [
+      {
+        "httpPathPrefix": "/DetectUnexpectedOtelMetrics/app2",
+        "connectionString": "InstrumentationKey=12345678-0000-0000-0000-0FEEDDADBEEF;IngestionEndpoint=http://host.testcontainers.internal:6060/"
+      },
+      {
+        "httpPathPrefix": "/DetectUnexpectedOtelMetrics/app3",
+        "connectionString": "InstrumentationKey=87654321-0000-0000-0000-0FEEDDADBEEF;IngestionEndpoint=http://host.testcontainers.internal:6060/"
+      }
+    ]
```
Comment (trask) [inline smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/resources/applicationinsights.json:29]: does this test need the extra complexity of role overrides?

File: smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DetectUnexpectedOtelMetricsTest.java (hunk 1 above)
Comment (heyams) [reply smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DetectUnexpectedOtelMetricsTest.java:40]: yes, i just used it to filter out these expected metrics and to check anything that is unexpected from otel

File: smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/resources/applicationinsights.json (hunk 2 above)
Comment (heyams) [reply smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/resources/applicationinsights.json:29]: initially, i was thinking of checking there is any rolename or ikey not being overwritten.

File: smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/fakeingestion/MockedAppInsightsIngestionServer.java [hunk 3]
```diff
@@ -221,6 +222,30 @@ public List<Envelope> waitForItemsInOperation(
...
+                if (md.getProperties().containsKey("_MS.MetricId")) {
+                  return false;
+                }
+                return condition.test(input);
+              }
+            },
+            1,
+            10,
+            TimeUnit.SECONDS);
+    return items;
```
Comment (trask) [inline smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/fakeingestion/MockedAppInsightsIngestionServer.java:246]: can inline `items`

File: smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DetectUnexpectedOtelMetricsTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+@UseAgent
+abstract class DetectUnexpectedOtelMetricsTest {
+
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+
+  @Test
+  @TargetUri("/app")
+  void testApp() throws Exception {
+    // no other otel metrics other than EXPECTED_OTEL_METRICS, expect an TimeoutException being
+    // thrown
```
Comment (trask) [inline smoke-tests/apps/DetectUnexpectedOtelMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/DetectUnexpectedOtelMetricsTest.java:30]:
```suggestion
    // verify no unexpected otel metrics, expect an TimeoutException being thrown
```

**Review by trask — APPROVED** [review]

**Pull request merged**